adam-core/src/main/java/org/bdgenomics/adam/io/InterleavedFastqInputFormat.java

```diff
@@ -246,15 +289,30 @@ protected boolean lowLevelFastqRead(Text readName, Text value) throws IOExceptio
                 throw new RuntimeException("unexpected fastq record didn't start with '@' at " + makePositionMessage() + ". Line: " + readName + ". \n");
 
             value.append(readName.getBytes(), 0, readName.getLength());
+            int sequenceLines;
+            
+            //check if getLineInfo has already been called on this read
+            if(readLines==-2){
+                //it has not yet been called
+                sequenceLines = getLineInfo();
+            }else{
+                //it has been called. Take line info from readLines and set it to -2.
+                sequenceLines=readLines;
+                readLines=-2;
+            }
 
             // sequence
+            for(int x = 0; x<sequenceLines; x++){
```

Comment:
This is a small detail, but in general, in scala it is more efficient to do `(0 to sequenceLines).foreach(x => {` instead of `for(int x = 0; x<sequenceLines; x++) {`